### PR TITLE
Add several new features to processor

### PR DESCRIPTION
**Pull request by mats-claassen**
Some of the new features:

* Aggregate data into reports
* Support multiple cameras
* Use local timezone
* Refactor configs and endpoints to access and update them
* Store more information in log files
* Add ability to upload screenshots to S3
* Add email notifications

TODO:
- [ ] Update Readme

### CODE REVIEW

**Review by renzodgc — APPROVED** [review]
Fileswise looks good to me. I'd wait for Mohsen's approval though

File: amd64-usbtpu.Dockerfile
Comment (emma-w-dev) [inline amd64-usbtpu.Dockerfile:92]: I guess pyhumps shouldn't be in this _apt purge_ line. it gives "E: Unable to locate package pyhumps" error.

File: README.md
Comment (emma-w-dev) [inline README.md:171]:
```suggestion
docker run -it --privileged -p HOST_PORT:8000 -v "$PWD/data":/repo/data -v "$PWD/config-coral.ini":/repo/config-coral.ini -e TZ=`./timezone.sh` neuralet/smart-social-distancing:latest-coral-dev-board
```

File: README.md
Comment (emma-w-dev) [inline README.md:180]:
```suggestion
docker run -it --privileged -p HOST_PORT:8000 -v "$PWD/data":/repo/data -v "$PWD/config-coral.ini":/repo/config-coral.ini -e TZ=`./timezone.sh` neuralet/smart-social-distancing:latest-amd64
```

File: config-coral.ini
Comment (emma-w-dev) [inline config-coral.ini:25]: I tried to run amd64-usbtpu docker and frontend. On the browser, there are requests such as this path: "filename: /static/gstreamer/default/playlist.m3u8" 
but I found that there are another similar path with "cam1" instead of "default". I changed this Id to "default" and I could see the videos. I think a modification is required in the paths, whether in the frontend requests or here in the configs.

Comment (mats-claassen) [reply config-coral.ini:25]: Yes, you are right. We should change one of them or make it clear in the Readme. The path to the video includes the camera ID (to allow multiple cameras). But the frontend is still pointing at a hardcoded 'default' camera (which is set in web_gui.py). Ideally the frontend would need to get the config from the processor and use the IDs defined there. but for now I can change the camera IDs to 'default' in these config files

File: config-jetson.ini
Comment (emma-w-dev) [inline config-jetson.ini:37]:
```suggestion
; ssd_mobilenet_v2_coco trained by neuralet on softbio dataset
```
as we don't have softbio model for the Jetson TX2 yet.

File: coral-dev-board.Dockerfile
Comment (emma-w-dev) [inline coral-dev-board.Dockerfile:71]: I had an error on building this line on pandas installation
Selection_015

File: config-jetson.ini
Comment (mats-claassen) [reply config-jetson.ini:37]: You are just changing the comment there. I guess we could instead add a note that that model is available only on Nano. And we could set the default value to the coco model

File: coral-dev-board.Dockerfile
Comment (mats-claassen) [reply coral-dev-board.Dockerfile:71]: Haven't seen that. I suppose it only happened on Coral? Did you happen to retry it just to check that it is not a temporal issue?
I can't test this on a Coral at the moment

File: config-jetson.ini
Comment (emma-w-dev) [reply config-jetson.ini:37]: sorry, wrong line, I wanted to suggest on name key.

File: config-jetson.ini
Comment (emma-w-dev) [inline config-jetson.ini:38]:
```suggestion
Name = ssd_mobilenet_v2_coco
```

File: coral-dev-board.Dockerfile
Comment (emma-w-dev) [reply coral-dev-board.Dockerfile:71]: yeah, I retried but the same happened.

**Comment by mhejrati** [comment]
@emma-w-dev did you have any comments after the last changes? Does this now pass the tests?

**Comment by mats-claassen** [comment]
Hey @mhejrati I have no Coral around and haven't been able to test the failure to build the Coral Dockerfile. It would be good if someone else could look into it.

**Comment by undefined-references** [comment]
@mhejrati I tested on amd64-tpu and jetsons and it was fine after my comments, coral is not okay yet. @JsonSadler is working on it.

File: coral-dev-board.Dockerfile
Comment (undefined-references) [reply coral-dev-board.Dockerfile:71]: this seems to be a memory issue and is solved using increasing swap size. so the docker and installation don't have problem.

**Review by undefined-references — APPROVED** [review]

**Comment by mhejrati** [comment]
All tests are done.

**Pull request merged**
